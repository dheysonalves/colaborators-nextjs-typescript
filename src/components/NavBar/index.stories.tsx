import { ComponentMeta, ComponentStory } from '@storybook/react'
import NavBar from './index'

export default {
	title: 'Layout/NavBar',
	component: NavBar,
} as ComponentMeta<typeof NavBar>

export const Default: ComponentStory<typeof NavBar> = () => <NavBar />
