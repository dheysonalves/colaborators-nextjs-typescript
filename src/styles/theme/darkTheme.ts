export const theme = {
	dark: {
		brand: {
			blue: '#034AFD',
		},
		neutral: {
			background: '#F8FAF9',
			light: '#fff',
			black: '#34423D',
			neutral_2: '#F5FAF8',
			neutral_3: '#A3B8B0',
			neutral_5: '#587169',
		},
		primary: {
			default: '#22E0A1',
			green: '#B5F1DD',
		},
		feedback: {
			success: '#B5F1DD',
		},
		tertiary: {
			disabled: '#EAEFED',
		},
	},
}
