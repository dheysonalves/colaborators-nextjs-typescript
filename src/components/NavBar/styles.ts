import styled from '@emotion/styled'
import { Container } from '../utils/Flex/flex.styles'
import { theme } from '../../styles/theme/lightTheme'

export const Wrapper = styled(Container)`
	border: 1px solid ${theme.light.tertiary.disabled};
	justify-content: space-between;
	align-items: center;
`
export const RightBorder = styled.div`
	border-right: 1px solid ${theme.light.tertiary.disabled};
	padding: 5px 10px 5px 10px;

	@media (max-width: 1199px) {
		flex: 1;
		text-align: center;
		border-right: none;
		padding: 0px;
	}
`

export const LeftBorder = styled.div`
	border-left: 1px solid ${theme.light.tertiary.disabled};
	padding: 5px 10px 5px 10px;
	@media (max-width: 1199px) {
		flex: 0;
		border-left: none;
		padding: 0px;
	}
`
