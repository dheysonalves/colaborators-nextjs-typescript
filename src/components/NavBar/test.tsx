import React from 'react'

import { ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import { theme } from '../../styles/theme/theme'

import NavBar from '.'

global.window.matchMedia = jest.fn().mockReturnValue({
	matches: true,
	addListener: () => {
		return
	},
	removeListener: () => {
		return
	},
})

const setup = () => {
	const utils = render(
		<ThemeProvider theme={theme}>
			<NavBar />
		</ThemeProvider>,
	)
	return {
		...utils,
	}
}

describe('<NavBar />', () => {
	it('should render the heading', () => {
		beforeEach(() => {
			window.matchMedia('(max-width: 1199px)')
		})
		const { container } = setup()

		expect(container.firstChild).toMatchSnapshot()
	})
})
