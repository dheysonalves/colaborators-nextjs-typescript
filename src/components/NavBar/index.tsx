import React from 'react'
import UserDetails from '../UserDetails'
import * as S from './styles'
import Media from 'react-media'

const NavBar = () => {
	return (
		<S.Wrapper>
			<Media
				query="(max-width: 1199px)"
				render={() => (
					<>
						<S.LeftBorder>
							<UserDetails name="Dheyson Alves" status="Testing" />
						</S.LeftBorder>
						<S.RightBorder>
							<img alt="logo" src="/img/brand.svg" width={32} height={32} />
						</S.RightBorder>
					</>
				)}
			/>
			<Media
				query="(min-width: 1200px)"
				render={() => (
					<>
						<S.RightBorder>
							<img alt="logo" src="/img/brand.svg" width={32} height={32} />
						</S.RightBorder>
						<S.LeftBorder>
							<UserDetails name="Dheyson Alves" status="Testing" />
						</S.LeftBorder>
					</>
				)}
			/>
		</S.Wrapper>
	)
}

export default NavBar
