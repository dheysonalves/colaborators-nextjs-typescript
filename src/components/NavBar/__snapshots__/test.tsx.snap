// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render the heading 1`] = `
<div
  class="css-1s162ov"
>
  <div
    class="css-7zkc34"
  >
    <div
      class="css-89ps4g"
    >
      <div
        class="css-t2lkvl"
      >
        <p
          class="css-1m4s1rb"
        >
          DA
        </p>
      </div>
      <div
        class="css-1s9w40j"
      >
        <h1
          class="css-a8r7il"
        >
          Dheyson Alves
        </h1>
        <h2
          class="css-1xrsu0h"
        >
          Testing
        </h2>
      </div>
    </div>
  </div>
  <div
    class="css-glkuc8"
  >
    <img
      alt="logo"
      height="32"
      src="/img/brand.svg"
      width="32"
    />
  </div>
  <div
    class="css-glkuc8"
  >
    <img
      alt="logo"
      height="32"
      src="/img/brand.svg"
      width="32"
    />
  </div>
  <div
    class="css-7zkc34"
  >
    <div
      class="css-89ps4g"
    >
      <div
        class="css-t2lkvl"
      >
        <p
          class="css-1m4s1rb"
        >
          DA
        </p>
      </div>
      <div
        class="css-1s9w40j"
      >
        <h1
          class="css-a8r7il"
        >
          Dheyson Alves
        </h1>
        <h2
          class="css-1xrsu0h"
        >
          Testing
        </h2>
      </div>
    </div>
  </div>
</div>
`
